src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,603 @@
+            sensorValue = sensorValue>>8;
+            iCount++;
+        } while(iCount < byteCount);
+
+        valueByteCount += 1 + byteCount;
+    }
+
+    exMessage[EXTEL_HEADER_TYPE_LEN] = 0x46+valueByteCount;
+    exMessage[(exMessage[EXTEL_HEADER_TYPE_LEN] & 0x3F) + 1] = calcCRC8(&exMessage[EXTEL_HEADER_TYPE_LEN], (exMessage[EXTEL_HEADER_TYPE_LEN]&0x3F));
+
+    return lastItem;
+}
+
+
+
+void createExBusMessage(uint8_t *exBusMessage, uint8_t *exMessage, uint8_t exBusID)
+{
+    uint16_t crc16;
+
+    exBusMessage[EXBUS_HEADER_PACKET_ID] = exBusID;
+    exBusMessage[EXBUS_HEADER_SUBLEN] = (exMessage[EXTEL_HEADER_TYPE_LEN] & 0x3F) + 2;    // +2: startbyte & CRC8
+    exBusMessage[EXBUS_HEADER_MSG_LEN] = 8 + exBusMessage[EXBUS_HEADER_SUBLEN];
+
+    crc16 = calcCRC16(exBusMessage,exBusMessage[EXBUS_HEADER_MSG_LEN] - JETIEXBUS_CRC_LEN);
+    exBusMessage[exBusMessage[EXBUS_HEADER_MSG_LEN] - 2] = crc16;
+    exBusMessage[exBusMessage[EXBUS_HEADER_MSG_LEN] - 1] = crc16>>8;
+}
+
+
+
+void initJetiExBusTelemetry(telemetryConfig_t *initialTelemetryConfig)
+{
+    UNUSED(initialTelemetryConfig);
+
+    // Init Ex Bus Frame header
+    jetiExBusTelemetryFrame[EXBUS_HEADER_SYNC] = 0x3B;       // Startbytes
+    jetiExBusTelemetryFrame[EXBUS_HEADER_REQ] = 0x01;
+    jetiExBusTelemetryFrame[EXBUS_HEADER_DATA_ID] = 0x3A;    // Ex Telemetry
+
+
+    // Init Ex Telemetry header
+    jetiExTelemetry[EXTEL_HEADER_SYNC] = 0x9F;              // Startbyte
+    jetiExTelemetry[EXTEL_HEADER_USN_LB] = 0x1E;            // Serial Number 4 Byte
+    jetiExTelemetry[EXTEL_HEADER_USN_HB] = 0xA4;
+    jetiExTelemetry[EXTEL_HEADER_LSN_LB] = 0x00;            // increment by telemetry count (%16) > only 15 values per device possible
+    jetiExTelemetry[EXTEL_HEADER_LSN_HB] = 0x00;
+    jetiExTelemetry[EXTEL_HEADER_RES] = 0x00;               // reserved, by default 0x00
+
+    return;
+}
+
+
+void checkJetiExBusTelemetryState(void)
+{
+    return;
+}
+
+void handleJetiExBusTelemetry(void)
+{
+    if (jetiExBusRequestState == EXBUS_STATE_RECEIVED) {
+        jetiExSensors[EX_VOLTAGE].value = vbat;
+        jetiExSensors[EX_CURRENT].value = amperage;
+        jetiExSensors[EX_ALTITUDE].value = BaroAlt;
+        jetiExSensors[EX_CAPACITY].value = mAhDrawn;
+        // switch to TX mode
+        if (uartTotalRxBytesWaiting(jetiExBusPort) == 0) {
+            serialSetMode(jetiExBusPort, MODE_TX);
+            jetiExBusTransceiveState = EXBUS_TRANS_TX;
+            sendJetiExBusTelemetry(jetiExBusSerialFrame[JETIEXBUS_FRAME_OFFSET + EXBUS_HEADER_PACKET_ID]);
+            jetiExBusRequestState = EXBUS_STATE_PROCESSED;
+        }
+    }
+
+    // check the state if transmit is ready
+    if (jetiExBusTransceiveState == EXBUS_TRANS_TX_READY) {
+        uartPort_t *s = (uartPort_t*)jetiExBusPort;
+        if (uartTotalTxBytesFree(jetiExBusPort) == (s->port.txBufferSize - 1)) {     // workaround for 'isUartTransmitBufferEmpty()'
```

Comment:
There is quite small window (char time is ~80us) when this can happen. Telemetry check is done at 250Hz (4000us), so the probability of collision is at most 2%.

Serial driver should implement TxEmpty check, but it is separate issue from this PR (other halfduplex telemetries have similar problem).